MaterialLib/SolidModels/Lubby2-impl.h

```diff
@@ -0,0 +1,269 @@
+                                                           3);
+
+    // Different solvers are available for the solution of the local system.
+    // TODO Make the following choice of linear solvers available from the
+    // input file configuration:
+    //      K_loc.partialPivLu().solve(-res_loc);
+    //      K_loc.fullPivLu().solve(-res_loc);
+    //      K_loc.householderQr().solve(-res_loc);
+    //      K_loc.colPivHouseholderQr().solve(res_loc);
+    //      K_loc.fullPivHouseholderQr().solve(-res_loc);
+    //      K_loc.llt().solve(-res_loc);
+    //      K_loc.ldlt().solve(-res_loc);
+
+    {  // Local Newton solver
+        using LocalResidualVector =
+            Eigen::Matrix<double, KelvinVectorSize * 3, 1>;
+
+        LocalJacobianMatrix K_loc;
+        auto const update_residual = [&](LocalResidualVector& residual) {
+            CalculateResidualBurgers(dt, epsd_i, sigd_j, state.eps_K_j,
+                                     state.eps_K_t, state.eps_M_j,
+                                     state.eps_M_t, residual, state);
+        };
+
+        auto const update_jacobian = [&](LocalJacobianMatrix& jacobian) {
+            CalculateJacobianBurgers(
+                t, x, dt, jacobian, sig_eff, sigd_j, state.eps_K_j,
+                state);  // for solution dependent Jacobians
+        };
+
+        auto const update_solution = [&](LocalResidualVector const& increment) {
+            // increment solution vectors
+            sigd_j.noalias() += increment.template block<KelvinVectorSize, 1>(
+                KelvinVectorSize * 0, 0);
+            state.eps_K_j.noalias() +=
+                increment.template block<KelvinVectorSize, 1>(
+                    KelvinVectorSize * 1, 0);
+            state.eps_M_j.noalias() +=
+                increment.template block<KelvinVectorSize, 1>(
+                    KelvinVectorSize * 2, 0);
+
+            // Calculate effective stress and update material properties
+            sig_eff = MaterialLib::SolidModels::Invariants<
+                KelvinVectorSize>::equivalentStress(sigd_j);
+            UpdateBurgersProperties(t, x, sig_eff * state.GM, state);
+        };
+
+        // TODO Make the following choice of maximum iterations and convergence
+        // criteria available from the input file configuration:
+        const int maximum_iterations(20);
+        const double tolerance(1.e-10);
+
+        auto newton_solver =
+            NewtonRaphson<decltype(linear_solver), LocalJacobianMatrix,
+                          decltype(update_jacobian), LocalResidualVector,
+                          decltype(update_residual), decltype(update_solution)>(
+                linear_solver, update_jacobian, update_residual,
+                update_solution, maximum_iterations, tolerance);
+
+        auto const success_iterations = newton_solver.solve(K_loc);
+
+        if (!success_iterations)
+            OGS_FATAL("The local Newton method did not succeed.");
```

Comment:
the failure should be informed to a user so that they can, e.g. decrease a time step size, to get convergence.